data_diff/diff_tables.py

```diff
@@ -99,38 +96,46 @@ def segment_by_checkpoints(self, checkpoints: List[DbKey]) -> List["TableSegment
 
         return tables
 
-        ## Calculate checksums in one go, to prevent repetitive individual calls
-        # selects = [t._make_select(columns=[Checksum(self._relevant_columns)]) for t in tables]
-        # res = self.database.query(Select(columns=selects), list)
-        # checksums ,= res
-        # assert len(checksums) == len(checkpoints) + 1
-        # return [t.new(_checksum=checksum) for t, checksum in safezip(tables, checksums)]
-
-    def new(self, _count=None, _checksum=None, **kwargs) -> "TableSegment":
+    def new(self, **kwargs) -> "TableSegment":
         """Using new() creates a copy of the instance using 'replace()', and makes sure the cache is reset"""
-        return self.replace(_count=None, _checksum=None, **kwargs)
-
-    @property
-    def count(self) -> int:
-        if self._count is None:
-            self._count = self.database.query(self._make_select(columns=[Count()]), int)
-        return self._count
+        return self.replace(**kwargs)
 
     @property
     def _relevant_columns(self) -> List[str]:
-        return (
-            [self.key_column]
-            + ([self.update_column] if self.update_column is not None else [])
-            + list(self.extra_columns)
+        extras = set(self.extra_columns)
+        if self.update_column:
+            extras.add(self.update_column)
+
+        return [self.key_column] + list(sorted(extras))
+
+    def count(self) -> Tuple[int, int]:
+        return self.database.query(self._make_select(columns=[Count()]), int)
+
+    def count_and_checksum(self) -> Tuple[int, int]:
+        start = time.time()
+        count, checksum = self.database.query(
+            self._make_select(columns=[Count(), Checksum(self._relevant_columns)]), tuple
         )
+        duration = time.time() - start
+        if duration > RECOMMENDED_CHECKSUM_DURATION:
+            logger.warn(f"Checksum is taking longer than expected ({duration:.2f}s). ")
```

Comment:
We need to offer a suggestion here: Increase the `bisection_factor`

We could have this same warning for `get_values` if it takes more than this duration, then `bisection_threshold` might be too high.

Reply:
It's better to have a suggestion, but I don't know what it is yet. I need to run some tests to know what would actually help.

Reply:
Increasing the `bisection_factor` means that the key ranges are going to be smaller, which means that individual queries should be faster 👍🏻

Reply:
I don't know why we should be worried about that, as it won't increase the total run-time.

Actually, now that I think of it, I'm not sure what's the point of this warning.

Reply:
The reason we need to be worried about it is that it's bad form to do queries on a relational databases that take several seconds. They take up a lot of capacity.